igc.h

```diff
@@ -161,4 +174,205 @@ class IgcFormat : public Format
     }
   };
 };
+  /*
+   * Extra data
+   *
+   * This includes any data recorded as part of an extension,
+   * which will be defined in the IGC file's I record,
+   * and present in individual B records.
+  */
+
+struct igc_fs_flags_t {
+  igc_fs_flags_t() :
+  has_igc_exts(0),
+  enl(0),
+  tas(0),
+  vat(0),
+  oat(0),
+  trt(0),
+  gsp(0),
+  fxa(0),
+  gfo(0),
+  siu(0),
+  acz(0)
+
+  {}
+
+  bool has_igc_exts;
+  bool enl;
+  bool tas;
+  bool vat;
+  bool oat;
+  bool trt;
+  bool gsp;
+  bool fxa;
+  bool gfo;
+  bool siu;
+  bool acz;
+};
+
+class ExtensionDefinition {
+public:
+  bool exists;  // Superfluous flag?
+  QString name; // Name of extension (ENL,FXA,TAS, etc)
+  short start;  // Start character of this data in B records
+  short end;    // End character of this data in B records
+  ExtensionDefinition (const QString name = "", short start=0, short end=0) :
+    exists(!name.isEmpty()),
+    name(name),
+    start(start),
+    end(end)
+  {}
+  void dump() {
+    printf("Extension data (%s):\n", this->name.toUtf8().constData());
+    printf("  Name: %s\n", this->name.toUtf8().constData());
+    printf("  Start: %i\n", this->start);
+    printf("  End: %i\n", this->end);
+  }
+};
+
+class IgcMetaData {
+private:
+  QMap<QString, ExtensionDefinition*> extensions;
+public:
+  igc_fs_flags_t flags;
+  IgcMetaData() {
```

Comment:
this ctor is unnecessary.  igc_fs_flags_t will be initailzied by it's ctor (or default initiailizers) anyway.